### PR TITLE
some improvements

### PR DESCRIPTION
**Pull request by caarlos0**
simplifed some code, changed get comments to accept a GET request, added server timeouts, improved table creation (with limits for the fields and create if not exists).

refs #29 #4 #34 #31

### CODE REVIEW

**Comment by adtac** [comment]
@caarlos0 thanks for the PR! I'm making a couple of changes on master, so please hold off merging from master for a while :)

**Comment by caarlos0** [comment]
@adtac no problem!
I'm also willing to add github authentication, what you think about that?

**Comment by adtac** [comment]
@caarlos0 Great, I'm done (I was adding a gitattributes file).

About https://github.com/adtac/commento/pull/36/commits/a0b607cbff517cdf64d255e23bfe8da50338cd97:
 - the `CREATE TABLE IF NOT NOT EXISTS` patch was fixed in https://github.com/adtac/commento/pull/27 :( Sorry about that.

**Comment by adtac** [comment]
Sorry, pressed Ctrl+Enter too early. I've edited the above comment :)

**Comment by adtac** [comment]
Personally, I don't like merge commits. With rebases, we can maintain a linear history of commits in master.

File: http.go [hunk 1]
```diff
@@ -64,12 +64,6 @@ func getCommentsHandler(w http.ResponseWriter, r *http.Request) {
 	var err error
 
 	result := resultContainer{Success: true}
-	if r.Method != "POST" {
```
Comment (GaiusMartius) [inline http.go]: +1

We should use a simple/fast router so we can isolate the routes by verb. Julienschmidts is httprouter is the fastest/simplest one.

adtac should approve this PR so I can start on that.

File: http.go [hunk 2]
```diff
@@ -68,8 +68,6 @@ func getCommentsHandler(w http.ResponseWriter, r *http.Request) {
 	if err != nil {
 		emit(err)
 	}
-
-end:
```
Comment (GaiusMartius) [inline http.go]: goto = wtf.

File: http.go (hunk 1 above)
Comment (adtac) [inline http.go]: @GaiusMartius sure, just waiting for the amended push from @caarlos0 :)

**Comment by caarlos0** [comment]
@adtac done, also removed the logger changes and referenced the issues open.

File: database.go [hunk 3]
```diff
@@ -12,9 +12,9 @@ func loadDatabase(dbFile string) error {
 	}
 	statement := `
 		CREATE TABLE IF NOT EXISTS comments (
-			url text not null,
-			name text not null,
-			comment text not null,
+			url varchar(2083) not null,
+			name varchar(200) not null,
+			comment varchar(10000) not null,
```
Comment (caarlos0) [inline database.go]: dont know if those sizes are good. the one I'm sure of is URL, which is set to the max possible url size.

Comment (adtac) [inline database.go]: 10000 comment limit is pretty good IMO. Someone posted the first hundred thousand digits of pi on the demo site :D

And I'm yet to meet someone with a name longer than 200 chars :P

Comment (caarlos0) [inline database.go]: yeah hahaha

**Comment by adtac** [comment]
@caarlos0 merging

**Comment by adtac** [comment]
@caarlos0 merged

@GaiusMartius you can pull the latest from master now :)

**Pull request merged**